plugins/keycloak-backend/src/lib/read.ts

```diff
@@ -66,11 +72,17 @@ export const parseUser = (
   spec: {
     profile: {
       email: user.email,
-      displayName: [user.firstName, user.lastName].filter(Boolean).join(' '),
+      displayName:
+        user.firstName || user.lastName
+          ? `${user.firstName} ${user.lastName}`
+          : user.username,
     },
-    memberOf: groups
-      .filter(g => g.members?.includes(user.username!))
-      .map(g => g.name!),
+    memberOf: groups.reduce((acc, g) => {
+      if (g.members?.includes(user.username!)) {
+        acc.push(g.name!);
+      }
+      return acc;
+    }, [] as string[]),
```

Comment:
How is this related to the merit of this PR? It just makes the code less readable. Performance improvements are really unmeasurable here.

Reply:
I will remove this change.